plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -202,174 +218,208 @@ instance uni ~ DefaultUni => ToBuiltinMeaning uni DefaultFun where
     toBuiltinMeaning Blake2b_256 =
         makeBuiltinMeaning
             Hash.blake2b
-            mempty -- TODO: budget. To be replace with: (runCostingFunOneArgument . paramBlake2b)
+            (runCostingFunOneArgument . paramBlake2b)
     toBuiltinMeaning VerifySignature =
         makeBuiltinMeaning
             (verifySignature @EvaluationResult)
             (runCostingFunThreeArguments . paramVerifySignature)
-    toBuiltinMeaning EqualsByteString =
-        makeBuiltinMeaning
-            ((==) @BS.ByteString)
-            (runCostingFunTwoArguments . paramEqualsByteString)
-    toBuiltinMeaning LessThanByteString =
-        makeBuiltinMeaning
-            ((<) @BS.ByteString)
-            (runCostingFunTwoArguments . paramLessThanByteString)
-    toBuiltinMeaning LessThanEqualsByteString =
-        makeBuiltinMeaning
-            ((<=) @BS.ByteString)
-            (runCostingFunTwoArguments . paramLessThanEqualsByteString)
-    toBuiltinMeaning IfThenElse =
-       makeBuiltinMeaning
-            (\b x y -> if b then x else y)
-            (runCostingFunThreeArguments . paramIfThenElse)
+    -- Strings
     toBuiltinMeaning AppendString =
         makeBuiltinMeaning
             ((<>) :: Text -> Text -> Text)
-            mempty  -- TODO: budget.
+            (runCostingFunTwoArguments . paramAppendByteString)
```

Comment:
```suggestion
            (runCostingFunTwoArguments . paramAppendString)
```

Reply:
> I only spotted one problem.
> 
> I don't fully get the new types you're adding for cost models. Main question: will they affect the names that go in the protocol parameters? i.e. do we need to pin them down now?

Yes, these will appear in the parameters, but the ones we actually use at the moment appear in `builtinCostModel.json` (maybe in truncated form, obtained using all this fancy Aeson deriving stuff) which I'm kind of regarding as being the 
source of truth at the moment. For instance, I stuck in some code for four- and five-argument builtins because it seemed a bit odd jumping from three to six, but we don't need them at the moment.  There are one or two names of specific models that we don't use either (`ModelTwoArgumentsLinearInX`, for example).  Presumably if we ever add new builtins that do use these things that'll be a sufficiently big change that we'll get the opportunity to update the protocol parameters too?  Maybe I'm wrong about that.

So if I've got this right, the names of the protocol parameters are derived from the stuff in `builtinCostModel.json` where we do that business of making a name out of the path through the tree, so it's only the stuff in there that we care about; and I think that everything in there is of the form we'll need when we add the missing R stuff, so we shouldn't need to change it (I hope).

Reply:
> 

The dangers of cut and paste.  Thanks for pointing that out.